units/commit_queue.go

```diff
@@ -66,6 +66,32 @@ func NewCommitQueueJob(env evergreen.Environment, queueID string, id string) amb
 	return job
 }
 
+func (j *commitQueueJob) TryUnstick(cq *commitqueue.CommitQueue) {
+	//unstuck the queue if the patch is done.
+	nextItem := cq.Next()
+	if nextItem != nil {
+		patch, err := patch.FindOne(patch.ById(patch.NewId(nextItem.Issue)).WithFields(patch.FinishTimeKey))
+		if err != nil {
+			j.AddError(errors.Wrapf(err, "error determining if patch is done for %s", j.QueueID))
+		}
+
+		//patchisdone
+		if !patch.FinishTime.IsZero() {
+			j.dequeue(cq, nextItem)
+			grip.Info(message.Fields{
+				"source":             "commit queue",
```

Comment:
Do you think we should still add that if we run this every minute, as opposed to every hour? It should never be more than a minute since the patch is finished.

Reply:
that's a sound argument.

Reply:
Is there a possibility of something going wrong, and the patch not dequeueing, and would it be helpful then?

Reply:
Perhaps, though it would probably need to be a db error, which is rare. There's also the possibility of a job backup.

Reply:
That's a good point. I'll add it.

Reply:
done. I can't think of a better name, but timeSincePatchDone seems awkward. Thoughts?

Reply:
can't think of anything better